src/convertToObject.js

```diff
@@ -6,7 +6,29 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString.length === 0) {
+    return {};
+  }
+
+  // debugger;
+
+  const getArrayFromString = sourceString.split(';');
+
+  const formateString = getArrayFromString
+    .filter((arrString) => arrString.trim().length > 0)
+    .map((arrString) => {
+      const [key, value] = arrString.split(':').map((part) => part.trim());
+
+      return {
+        [key]: value
+          .split(', ') // добавив тут пробіл і все спрацювало, ну і шляпа
+          .map((v) => v.trim())
+          .join(','),
+      };
+    });
+  const outPutObj = Object.assign({}, ...formateString);
```

Comment:
[CODE PRACTICE] - Avoid using loops in this task, use iteration methods instead. While `Object.assign` is not a loop, you could consider using the spread operator to merge objects within an array in a more modern and concise way.